functions/package.json

```diff
@@ -0,0 +1,28 @@
+{
+  "name": "functions",
+  "description": "Cloud Functions for Firebase",
+  "scripts": {
+    "lint": "eslint .",
+    "serve": "firebase emulators:start --only functions",
+    "shell": "firebase functions:shell",
+    "start": "npm run shell",
+    "deploy:functions": "firebase deploy --only functions",
+    "deploy:firestore": "firebase deploy --only firestore:rules",
+    "logs": "firebase functions:log"
+  },
+  "engines": {
+    "node": "16"
+  },
```

Comment:
This forces users to use node v16, unsure if it's necessary. I removed it locally to see if it would allow me to run node v17.4 and it seemed to be fine.

Reply:
it is necessary as I need to specify the node version in order to deploy the function. The latest version of node available on Google Cloud Platform is v16.